Added PIL + pathlib Tutorial (#5862)
python-pillow/Pillow: docs/handbook/tutorial.rst

        # ...do something to frame...

> Batch processing with Pathlib

Reviewer: ```suggestion
Batch processing with pathlib
```

-----------------------------

> This example uses PIL togehter with pathlib, in order to reduce the quality of all png images in a folder

Reviewer 2: ```suggestion
This example uses PIL together with pathlib, in order to reduce the quality of all png images in a folder
```

> This example uses PIL together with pathlib, in order to reduce the quality of all png images in a folder

Reviewer: ```suggestion
This example uses Pillow together with pathlib, in order to reduce the quality of all PNG images in a folder:
```

::

    from PIL import Image
    from pathlib import Path

>     def compressImg(filepath, verbose=False):

Reviewer 2: ```suggestion
    def compressImg(filepath):
```
This argument isn't used.

        file = filepath.stem

>         img = Image.open(filepath)
>         if img.mode != 'RGB':
>             img = img.convert('RGB')
>         img.save(file + ".jpg",
>                 "JPEG",
>                 optimize=True,
>                 quality=80)

Reviewer 2: ```suggestion
        with Image.open(filepath) as img:
            if img.mode != 'RGB':
                img = img.convert('RGB')
            img.save(file + ".jpg",
                    "JPEG",
                    optimize=True,
                    quality=80)
```

Since Pillow 7.0, users should either use a context manager or call `close()` - https://pillow.readthedocs.io/en/stable/releasenotes/7.0.0.html#image-del

>     from PIL import Image
>     from pathlib import Path
> 
> 
>     def compressImg(filepath):
>         file = filepath.stem
>         with Image.open(filepath) as img:
>             if img.mode != 'RGB':
>                 img = img.convert('RGB')
>             img.save(file + ".jpg",
>                     "JPEG",
>                     optimize=True,
>                     quality=80)
>         return
> 
> 
>     base_directory = Path.cwd()
> 
>     for path in base_directory.iterdir():
>         if path.suffix == ".png":
>             print(path)
>             compressImg(path)

Reviewer: Format with Black and use a lowercase function name:
```suggestion
    from pathlib import Path
    from PIL import Image

    def compress_image(filepath):
        file = filepath.stem
        with Image.open(filepath) as img:
            if img.mode != "RGB":
                img = img.convert("RGB")
            img.save(file + ".jpg", "JPEG", optimize=True, quality=80)
        return

    base_directory = Path.cwd()

    for path in base_directory.iterdir():
        if path.suffix == ".png":
            print(path)
            compress_image(path)
```

    fp = TarIO.TarIO("Tests/images/hopper.tar", "hopper.jpg")
    im = Image.open(fp)

Batch processing
^^^^^^^^^^^^^^^^

Operations can be applied to multiple image files. For example, all PNG images
in the current directory can be saved as JPEGs at reduced quality.

::

>     import os

Reviewer 2: ```suggestion
    import glob
```

    from PIL import Image

    def compress_image(source_path, dest_path):
        with Image.open(source_path) as img:
            if img.mode != "RGB":
                img = img.convert("RGB")
            img.save(dest_path, "JPEG", optimize=True, quality=80)

>     paths = glob.glob(".png")

Reviewer 2: ```suggestion
    paths = glob.glob("*.png")
```
This needs an asterisk to work as intended.

    for path in paths:
        compress_image(path, path[:-4] + ".jpg")

Since images can also be opened from a ``Path`` from the ``pathlib`` module,

> the example could be modified to use ``pathlib`` instead of ``os``.

Reviewer 2: ```suggestion
the example could be modified to use ``pathlib`` instead of the ``glob``
module.
```

    fp = TarIO.TarIO("Tests/images/hopper.tar", "hopper.jpg")
    im = Image.open(fp)

Batch processing
^^^^^^^^^^^^^^^^

Operations can be applied to multiple image files. For example, all PNG images
in the current directory can be saved as JPEGs at reduced quality.

::

    import glob
    from PIL import Image

    def compress_image(source_path, dest_path):
        with Image.open(source_path) as img:
            if img.mode != "RGB":
                img = img.convert("RGB")
            img.save(dest_path, "JPEG", optimize=True, quality=80)

    paths = glob.glob("*.png")
    for path in paths:
        compress_image(path, path[:-4] + ".jpg")

Since images can also be opened from a ``Path`` from the ``pathlib`` module,
the example could be modified to use ``pathlib`` instead of the ``glob``
module.

::

    from pathlib import Path

    paths = Path(".").glob("*.png")
    for path in paths:

>         compress_image(path, filepath.stem + ".jpg")

Reviewer 2: ```suggestion
        compress_image(path, path.stem + ".jpg")
```
